internal/core/input.rs

```diff
@@ -172,6 +172,61 @@ pub struct KeyEvent {
     pub event_type: KeyEventType,
 }
 
+impl KeyEvent {
+    /// If a shortcut was pressed, this function returns `Some(Shortcut)`.
+    /// Otherwise it returns None.
+    pub fn shortcut(&self) -> Option<Shortcut> {
+        if self.modifiers.control && !self.modifiers.shift {
+            match self.text.as_str() {
+                "c" => Some(Shortcut::Copy),
+                "x" => Some(Shortcut::Cut),
+                "v" => Some(Shortcut::Paste),
+                "a" => Some(Shortcut::SelectAll),
+                "f" => Some(Shortcut::Find),
+                "s" => Some(Shortcut::Save),
+                "p" => Some(Shortcut::Print),
+                "z" => Some(Shortcut::Undo),
+                #[cfg(not(target_os = "macos"))]
+                "y" => Some(Shortcut::Redo),
+                "r" => Some(Shortcut::Refresh),
+                _ => None,
+            }
+        } else if self.modifiers.control && self.modifiers.shift {
+            match self.text.as_str() {
+                #[cfg(target_os = "macos")]
+                "z" => Some(Shortcut::Redo),
```

Comment:
Ctrl+shift+Z  is also redo on linux, i use that all the time.

Reply:
hmm. qt allows this on windows as well, but Windows itself does not.